agent/internal/agent.go

```diff
@@ -123,6 +127,10 @@ func (a *agent) Receive(ctx *actor.Context) error {
 		a.handleAPIRequest(ctx, msg)
 
 	case actor.PostStop:
+		if err := a.fluent.StopAndAwaitTermination(); err != nil {
```

Comment:
blocking: one last minor nit, if we're stopping because we failed to start fluent, this deref's a nil pointer and leads to a confusing-ish stack trace.